modules/javafx.controls/src/test/java/test/javafx/scene/control/MenuBarTest.java

```diff
@@ -114,10 +114,26 @@ protected void startApp(Parent root) {
     }
 
     @Test public void defaultConstructorButSetTrueFocusTraversable() {
-            menuBar.setFocusTraversable(true);
+        menuBar.setFocusTraversable(true);
         assertTrue(menuBar.isFocusTraversable());
     }
 
+    @Test public void testFocusOnEmptyMenubar() {
+        menuBar.setFocusTraversable(true);
+
+        AnchorPane root = new AnchorPane();
+        root.getChildren().add(menuBar);
+        startApp(root);
+
+        MenuBarSkin skin = (MenuBarSkin)menuBar.getSkin();
+        assertTrue(skin != null);
+
+        menuBar.getScene().getWindow().requestFocus();
+
+        int focusedIndex = MenuBarSkinShim.getFocusedIndex(skin);
+        assertEquals(focusedIndex, -1);
```

Comment:
the assert should be the other way round, expected value should be the first parameter :)

didn't notice on first read and now only when writing a test case against our argument, c&p'ed the test as-is, replaced the requestFocus with simulating the notification from traversalListener and was confused about its value being -1 

Here's that modified test method (requires test api in MenuBarSkin and shim):

    @Test public void testSimulateTraverseIntoOnEmptyMenubar() {
        menuBar.setFocusTraversable(true);

        AnchorPane root = new AnchorPane();
        root.getChildren().add(menuBar);
        startApp(root);

        MenuBarSkin skin = (MenuBarSkin)menuBar.getSkin();
        assertTrue(skin != null);

        // simulate notification from traversalListener 
        MenuBarSkinShim.setFocusedIndex(skin, 0);
        int focusedIndex = MenuBarSkinShim.getFocusedIndex(skin);
        assertEquals(-1, focusedIndex);
    }

Reply:
Corrected the similar asserts in this file.

Thanks for suggestion on test. Yes. At best, we can simulate this with Shim.
I have added it now.